zebrad/src/commands/start/sync.rs

```diff
@@ -425,6 +445,26 @@ where
 
         Ok(())
     }
+
+    /// Get the depth of hash in the current chain.
+    ///
+    /// Returns None if the hash is not present in the chain.
+    /// TODO: handle multiple tips in the state.
```

Comment:
How do you expect the multiple tips to affect this code? The only thing I can think of is race conditions related to depths that are invalidated by updates to the state in the background, but I think we're always using the depth to test for the presence of a block, not to specifically locate it, so it doesn't seem like it would matter here.

Reply:
I'd like the state service to add a request that checks if a block is present in any chain.

Then we don't have to worry about the exact depth and race conditions.

(We might occasionally re-download some side-chains near the reorg limit, but they should be rejected or dropped pretty quickly. So that's not a correctness issue.)

Reply:
I opened #862 for this change.

Reply:
> Then we don't have to worry about the exact depth and race conditions.
>
> Then we don't have to worry about the exact depth and race conditions.

That's the thing I'm not seeing though, what race conditions are there and in what situations does the exact depth matter?

Reply:
Since there's a follow-on issue already, it seems like we could merge this in now and sort out this detail later?

Reply:
I replied in #862 - I think we are actually agreeing about the details, but I want to change the interface.